include/factory_data.h

```diff
@@ -28,10 +28,12 @@
  * Do not access fields directly. Use API functions only. */
 typedef struct {
   uint32_t _hardware;
-  uint32_t _serial_number;
+  uint8_t _mfg_serial_number[17];
```

Comment:
Why not just `serial_number`? The `mfg` part seems redundant since the whole thing is "factory data".

Why 17 bytes?